include/fault.h

```diff
@@ -0,0 +1,103 @@
+#ifndef FAULT_H
+#define FAULT_H
+
+#include "ultra64.h"
+#include "libc/stdarg.h"
+#include "libc/stdint.h"
+#include "io/controller.h"
+#include "stack.h"
+
+// These are the same as the 3-bit ansi color codes
+#define FAULT_COLOR_BLACK      0
+#define FAULT_COLOR_RED        1
+#define FAULT_COLOR_GREEN      2
+#define FAULT_COLOR_YELLOW     3
+#define FAULT_COLOR_BLUE       4
+#define FAULT_COLOR_MAGENTA    5
+#define FAULT_COLOR_CYAN       6
+#define FAULT_COLOR_WHITE      7
+// Additional color codes
+#define FAULT_COLOR_DARK_GRAY  8
+#define FAULT_COLOR_LIGHT_GRAY 9
+
+#define FAULT_COLOR_STRINGIFY(s) #s
+#define FAULT_COLOR_EXPAND_AND_STRINGIFY(s) FAULT_COLOR_STRINGIFY(s)
+
+#define FAULT_ESC '\x1A'
+#define FAULT_COLOR(n) "\x1A" FAULT_COLOR_EXPAND_AND_STRINGIFY(FAULT_COLOR_ ## n)
+
+
+#define NORMAL_RDRAM_END 0x80400000
+// Address at the end of normal RDRAM after which is room for a screen buffer
+#define FAULT_FB_ADDRESS (NORMAL_RDRAM_END - SCREEN_HEIGHT * SCREEN_WIDTH * sizeof(u16))
```

Comment:
Couple things here.

Considering that Fault_DisplayFrameBuffer implements it's framebuffer pointer as 
```c
fb = (void*)(PHYS_TO_K0(osMemSize) - SCREEN_HEIGHT * SCREEN_WIDTH * sizeof(u16));```

should we not express the address this way?

```c
#define FAULT_FB_ADDRESS (PHYS_TO_K0(0x400000) - SCREEN_HEIGHT * SCREEN_WIDTH * sizeof(u16))```


Second, the wording of the comment confused me for a bit, since the way it reads can make it seem like the framebuffer comes after the end of the hardware's 4MB RDRAM. Imo, the key thing to express here is that the framebuffer is reserved at the end of the 4MB already installed on the N64, since MM is expected to handle booting without an expansion pack installed.


Lastly, I truly despise the `NORMAL` in `NORMAL_RDRAM_END`. The word normal should only ever be used in the context of math and game options; for everything else you should be using something more descriptive.

Reply:
This is an already existing comment and define which I moved from another file to this header file. I don't really like it but it is used in sys_initial_check and I didn't want to mess what's going on there.
I guess I can yeet `NORMAL_RDRAM_END` and use the `PHYS_TO_K0(0x400000)` instead